spec/system/admin/bulk_product_update_spec.rb

```diff
@@ -227,51 +227,85 @@
+
+      # When I fill out variant details and hit update
+      fill_in "variant_display_name", with: "Case of 12 Bottles"
+      fill_in "variant_unit_value_with_description", with: "3 (12x250 mL bottles)"
+      fill_in "variant_display_as", with: "Case"
+      fill_in "variant_price", with: "4.0"
+      fill_in "variant_on_hand", with: "10"
+
+      click_button 'Save Changes', match: :first
+      expect(page.find("#status-message")).to have_content "Changes saved."
+
+      updated_variant = Spree::Variant.where(deleted_at: nil).last
+      expect(updated_variant.display_name).to eq "Case of 12 Bottles"
+      expect(updated_variant.unit_value).to eq 3000
+      expect(updated_variant.unit_description).to eq "(12x250 mL bottles)"
+      expect(updated_variant.display_as).to eq "Case"
+      expect(updated_variant.price).to eq 4.0
+      expect(updated_variant.on_hand).to eq 10
+
+      # Then I should see edit buttons for the new variant
+      expect(page).to have_selector "a.edit-variant", visible: true
     end
-    expect(page).to have_selector "tr.variant", count: 2
-    page.all('a.delete-variant', visible: true).first.click
-    expect(page).to have_selector "tr.variant", count: 1
 
-    # When I fill out variant details and hit update
-    fill_in "variant_display_name", with: "Case of 12 Bottles"
-    fill_in "variant_unit_value_with_description", with: "3 (12x250 mL bottles)"
-    fill_in "variant_display_as", with: "Case"
-    fill_in "variant_price", with: "4.0"
-    fill_in "variant_on_hand", with: "10"
+    context "handle the 'on_demand' variant case creation" do
+      # let(:variant) { create(:variant, product: p) }
 
-    click_button 'Save Changes', match: :first
-    expect(page.find("#status-message")).to have_content "Changes saved."
+      before do
+        p = Spree::Product.first
+        p.master.update_attribute(:on_hand, 5)
+        p.save
+        v1 = FactoryBot.create(:variant, product: p, is_master: false, on_hand: 4)
+        v2 = FactoryBot.create(:variant, product: p, is_master: false, on_demand: true)
+        p.variants << v1
+        p.variants << v2
+      end
+
+      it "when variant unit value is: '120'" do
+        fill_in "variant_unit_value_with_description", with: "120"
+        check "variant_on_demand"
+
+        click_button 'Save Changes', match: :first
+        expect(page.find("#status-message")).to have_content "Changes saved."
+      end
 
-    updated_variant = Spree::Variant.where(deleted_at: nil).last
-    expect(updated_variant.display_name).to eq "Case of 12 Bottles"
-    expect(updated_variant.unit_value).to eq 3000
-    expect(updated_variant.unit_description).to eq "(12x250 mL bottles)"
-    expect(updated_variant.display_as).to eq "Case"
-    expect(updated_variant.price).to eq 4.0
-    expect(updated_variant.on_hand).to eq 10
+      it "when variant unit value is: '120g'" do
+        fill_in "variant_unit_value_with_description", with: "120g"
```

Comment:
On this test case, the variant is actually not being updated. For example, if we change the price as well, with something like this:

`        fill_in "variant_price", with: "6.66"`

And then after saving changes, assert that the price changed with:

`        expect(page).to have_content "6.66"`

Then the test will fail, probably because of the `g` in `120g`.

This looks closely related to #9068; do you think we need a separate issue?

Reply:
I think this should all be handled by #9068 ;)